pkg/operator/reports.go

```diff
@@ -859,6 +923,50 @@ func (op *Reporting) queueDependentReportsForReport(report *metering.Report) err
 	return nil
 }
 
+func (op *Reporting) doReportsDependOnThisReport(report *metering.Report, reports []*metering.Report) bool {
+	// for each report in the namespace, find ones that depend on the report passed into the function.
+	for _, otherReport := range reports {
+		deps, err := op.getReportDependencies(otherReport)
+		if err != nil {
+			op.logger.Errorf("was unable to get dependencies for report: %s, while looking for dependent reports", report.Name)
+			return false
+		}
+		// If a Report has a dependency on the passed in report, we're done
+		for _, dep := range deps.Reports {
+			if dep.Name == report.Name {
+				break
+			}
+		}
+	}
+	// true means no other reports depend on this report
+	return true
+}
+
+func (op *Reporting) reportQueriesDependOnThisReport(reportQuery *metering.ReportQuery) bool {
+	// Look for all ReportQueries in the namespace
+	reportQueries, err := op.reportQueryLister.ReportQueries(reportQuery.Namespace).List(labels.Everything())
+	if err != nil {
+		op.logger.Errorf("was unable to list reports for report: %s, while looking for dependent queries")
+		return true
+	}
```

Comment:
I'm fine with making the call to list the ReportQueries in the same namespace of the `@reportQuery` parameter, but we handle this slightly different in the `op.doReportsDependOnThisReport` and pass both the target resource and the list of those resources in a namespace as parameters. I think I would prefer to make the call to list the resources we care about at the call site instead of in the helper method, but I'm fine with either implementation if we make it consistent.

Reply:
agreed fixed up now.